modules/javafx.swing/src/main/java/javafx/embed/swing/SwingNode.java

```diff
@@ -365,12 +364,17 @@ public JComponent getContent() {
      */
     private void setContentImpl(JComponent content) {
         if (lwFrame != null) {
-            rec.dispose();
-            Disposer.removeRecord(disposerRecRef);
-            rec = null;
+            if (rec != null) {
+                rec.dispose();
+                rec = null;
+	    }
             lwFrame = null;
         }
         if (content != null) {
+            if (disposerRecRef != null) {
+                Disposer.removeRecord(disposerRecRef);
+                disposerRecRef = null;
+            }
```

Comment:
I think this should be part of the first `if`, or an `if` by itself, and not here.  `lwFrame`, `rec` and `disposerRecRef` will all either be `null` or be set to some value.  The only case this may not happen is that after `lwFrame` is set, an exception occurs.  You could make it very safe, or just assume that when `lwFrame` is not `null` that all clean up actions should be taken -- if an exception did occur during set-up, then it's also fine that one can occur during clean-up, no need to make it more robust for situations that can't happen.

There really is no need to start second guessing yourself when the logic is solid.  Either all those variables are non-`null` and checking one of them is sufficient, or they are all `null` and checking each in turn is pointless (it is just confusing because programmers may assume now that is a situation that can actually occur).

Please revert to the old code.

Reply:
@hjohn I guess i have updated as per suggestion..Let me know if this is ok as per you..if yes, please approve..